src/force/CabanaPD_Force_PMB.hpp

```diff
@@ -161,6 +161,43 @@ class Force<ExecutionSpace, ForceModel<PMB, Elastic, ModelParams...>>
             double w = 0.25 * model.c * s * s * xi * vol( j );
             W( i ) += w;
             Phi += w * vol( i );
+
+
+            
+            // Calculate forces for virial stress
+            double fx_i = 0.0;
+            double fy_i = 0.0;
+            double fz_i = 0.0;
+
+            double rx, ry, rz;
+            getDistanceComponents( x, u, i, j, xi, r, s, rx, ry, rz );
+
+            // Check if the bond is broken
+            //if ( model.criticalStretch( i, j, r, xi ) && !nofail( i ) && !nofail( j ) )
+            //{
+            //    return;  // Skip this bond if broken
+            //}
+            
+            const double coeff = model.c * s * vol( j );
+            fx_i = coeff * rx / r;
+            fy_i = coeff * ry / r;
+            fz_i = coeff * rz / r;
+
+            //f( i, 0 ) += fx_i; //no point accumulating them for virial stress
+            //f( i, 1 ) += fy_i;
+            //f( i, 2 ) += fz_i;
+
+            auto virial_stress = particles.sliceVirialStress();
```

Comment:
You are allowed to slice in kernel, but for performance we normally do it outside (as with volume)

Reply:
Moved it outside with the volume